### PR TITLE
Add ESLint and JSCS

### LINKED ISSUE #2: Add jscs
Use jscs-preset-gulp

### LINKED ISSUE #3: Explicit eslint usage
To bring inline with gulp repos. Use eslint-config-gulp and update code style

### LINKED ISSUE #4: Update repo location references
In package.json, readme, etc

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #2 
Fixes #3 
Fixes #4

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,13 +1,6 @@
 {
-  "rules": {
-    "semi": [2, "always"],
-    "quotes": [2, "single"],
-    "strict": [2, "global"],
-    "handle-callback-err": 0,
-    "no-underscore-dangle": 0
-  },
+  "extends": "gulp",
   "env": {
```
Comment (phated) [inline .eslintrc]: these can be removed

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,8 @@
-sudo: false
 language: node_js
+
+sudo: false
```
Comment (phated) [inline .travis.yml:1]: I prefer sudo: false to be at the top

File: package.json [hunk 3]
```diff
@@ -4,11 +4,13 @@
   "description": "Guarantee a node-style async function is only executed once",
   "main": "index.js",
   "scripts": {
+    "lint": "eslint . && jscs *.js test/",
+    "pretest": "npm run lint",
     "test": "lab -cvL"
```
Comment (phated) [inline package.json]: remove linting from the lab running

**Comment by pdehaan** [comment]
Done.

**Pull request merged**